pkg/api/api.go

```diff
@@ -364,12 +363,12 @@ func requestDeferred(r *http.Request) (bool, error) {
 	return true, nil
 }
 
-func requestPostageBatchId(r *http.Request) ([]byte, error) {
+func requestPostageBatchId(r *http.Request) (swarm.BatchID, error) {
 	if h := strings.ToLower(r.Header.Get(SwarmPostageBatchIdHeader)); h != "" {
 		if len(h) != 64 {
 			return nil, errInvalidPostageBatch
 		}
-		b, err := hex.DecodeString(h)
+		b, err := swarm.HexToBatchID(h)
```

Comment:
I dont understand why this function is needed?

Original function is simple, readable enough. Secondly, it is doing the exact same thing.

Reply:
`swarm.HexToBatchID`  returns `BatchID` from provided hex string, while 
`hex.DecodeString` returns `[]byte` from provided hex string.